daemon/cluster/networks.go

```diff
@@ -87,9 +87,14 @@ func (c *Cluster) UpdateAttachment(target, containerID string, config *network.N
 	if !ok || attacher == nil {
 		return fmt.Errorf("could not find attacher for container %s to network %s", containerID, target)
 	}
+	if attacher.config != nil {
```

Comment:
Thanks. Given the receiver is not guarantee to still be there on the second call, this function could be waiting indefinitely. I will check if it is ok to make `attachWaitCh`  a buffered channel and add the logic to discard the duplicate config at the receiver to cover the case it is still alive to process it.